fflib/src/classes/fflib_QueryFactoryTest.cls

```diff
@@ -151,6 +151,45 @@ private class fflib_QueryFactoryTest {
 		System.assert( Pattern.matches('SELECT.*Email.*FROM.*',query), 'Expected Name field in query, got '+query);
 	}
 
+	@isTest
+	static void orderingSet(){
+		fflib_QueryFactory qf = new fflib_QueryFactory(Contact.SObjectType);
+		qf.selectField('name');
+		qf.selectField('email');
+		qf.setCondition( 'name = \'test\'' );
+		qf.setOrdering( new fflib_QueryFactory.Ordering('Contact','LastModifiedDate',fflib_QueryFactory.SortOrder.DESCENDING) );
+		
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering('LastModifiedDate', fflib_QueryFactory.SortOrder.DESCENDING, true);
+		
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering(Contact.LastModifiedDate, fflib_QueryFactory.SortOrder.DESCENDING, true);
+
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering('LastModifiedDate', fflib_QueryFactory.SortOrder.DESCENDING);
+
+		System.assertEquals(1,qf.getOrderings().size());
```

Comment:
**ACTION: Add assert failure messages.**
You could use the assert message to indicate the failing behaviour - particularly in this case where there are so many assertions.
```java
qf.setOrdering('LastModifiedDate', fflib_QueryFactory.SortOrder.DESCENDING);

System.assertEquals(1, qf.getOrderings().size(), 'Unexpected order size - setOrder should replace previous Orderings');
System.assertEquals(Contact.LastModifiedDate, qf.getOrderings()[0].getField(), 'Unexpected order field - should have been resolved from the field name' );
System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING, qf.getOrderings()[0].getDirection(), 'Unexpected order direction');
```